model/system_metrics.go

```diff
@@ -88,28 +92,103 @@ var (
 // Find searches the database for the system metrics object. The environment should
 // not be nil. Either the ID or full Info of the system metrics object needs to be
 // specified.
-func (result *SystemMetrics) Find(ctx context.Context) error {
-	if result.env == nil {
+func (sm *SystemMetrics) Find(ctx context.Context) error {
+	if sm.env == nil {
 		return errors.New("cannot find with a nil environment")
 	}
 
-	if result.ID == "" {
-		result.ID = result.Info.ID()
+	if sm.ID == "" {
+		sm.ID = sm.Info.ID()
 	}
 
-	result.populated = false
-	err := result.env.GetDB().Collection(systemMetricsCollection).FindOne(ctx, bson.M{"_id": result.ID}).Decode(result)
+	sm.populated = false
+	err := sm.env.GetDB().Collection(systemMetricsCollection).FindOne(ctx, bson.M{"_id": sm.ID}).Decode(sm)
 	if db.ResultsNotFound(err) {
-		return fmt.Errorf("could not find system metrics record in the database with id %s", result.ID)
+		return fmt.Errorf("could not find system metrics record in the database with id %s", sm.ID)
 	} else if err != nil {
-		return errors.Wrapf(err, "problem finding system metrics with id %s", result.ID)
+		return errors.Wrapf(err, "problem finding system metrics with id %s", sm.ID)
 	}
 
-	result.populated = true
+	sm.populated = true
 
 	return nil
 }
 
+// Append uploads a chunk of system metrics data to the offline blob storage bucket
+// configured for the system metrics and updates the metadata in the database to reflect
+// the uploaded data. The environment should not be nil.
+func (sm *SystemMetrics) Append(ctx context.Context, order int, data []byte) error {
+	if sm.env == nil {
+		return errors.New("cannot not append system metrics data with a nil environment")
+	}
+	if len(data) == 0 {
+		grip.Warning(message.Fields{
+			"collection": systemMetricsCollection,
+			"id":         sm.ID,
+			"message":    "append called with no system metrics data",
+		})
+		return nil
+	}
+
+	key := strconv.Itoa(order)
```

Comment:
I am a little skeptical about using "order" as the key names. When we stitch them together we would end up having to sort the slice of keys. Also, what happens if users give us two chunks of data with the same order? Can we maybe use timestamps like we do with buildlogger? We could then write a client that buffers the time series data and sends it in order.